pkg/loop/plugin_generic.go

```diff
@@ -0,0 +1,56 @@
+package loop
+
+import (
+	"context"
+
+	"github.com/hashicorp/go-plugin"
+	"google.golang.org/grpc"
+
+	"github.com/smartcontractkit/chainlink-relay/pkg/loop/internal"
+	"github.com/smartcontractkit/chainlink-relay/pkg/types"
+)
+
+// PluginGenericName is the name for [types.PluginGeneric]/[NewGRPCPluginGeneric].
+const PluginGenericName = "generic"
+
+func PluginGenericHandshakeConfig() plugin.HandshakeConfig {
+	return plugin.HandshakeConfig{
+		MagicCookieKey:   "CL_PLUGIN_GENERIC_MAGIC_COOKIE",
+		MagicCookieValue: "2ad981747cd86c4ab3e23170970020fd",
+	}
+}
+
+type GRPCPluginGeneric struct {
+	plugin.NetRPCUnsupportedPlugin
+
+	BrokerConfig
+
+	PluginServer types.PluginGeneric
```

Comment:
This should use a different type which converts the connection into a proper provider